### PR TITLE
bt: Add support for coredumps

### PR DESCRIPTION
**Pull request by BertalanD**
This commit adds support for reading backtraces from coredumps to `bt`.
This makes `bt` the command-line counterpart of CrashReporter, and will
be used for generating backtraces for crashed CI tests.

---
I can't figure out how to spawn `bt` nicely from `run-tests`, but let's kick off the review process with this commit.
My CI coredump branch can be seen [here](https://github.com/BertalanD/serenity/commits/ci-coredump). Don't hesitate to contact me if you have any suggestions regarding it.

### CODE REVIEW

File: Userland/Utilities/bt.cpp [hunk 1]
```diff
@@ -80,3 +76,60 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
...
+{
+    TRY(Core::System::pledge("stdio rpath"));
+
+    Core::ArgsParser args_parser;
+    String pid_or_coredump;
+    args_parser.set_general_help("View the backtrace of a specified process");
+    args_parser.add_positional_argument(pid_or_coredump, "PID or path to coredump", "pid-or-coredump");
+    args_parser.parse(arguments);
+
+    if (auto maybe_pid = pid_or_coredump.to_int<pid_t>(); maybe_pid.has_value())
```
Comment (awesomekling) [inline Userland/Utilities/bt.cpp:131]: This feels a bit rickety. What if I have a coredump file named "10"? :)

Comment (BertalanD) [reply Userland/Utilities/bt.cpp:131]: Then you disambiguate by running `bt ./10` :^)

But seriously, someone moving the `/tmp/coredump/<name>_<pid>_<timestamp>` file into a file named `10` would not be a normal use case. Or if we want to support that, we could `stat` for a file named "10" and ask the user to disambiguate if needed.

Comment (awesomekling) [reply Userland/Utilities/bt.cpp:131]: How about we avoid the guesswork with `bt -f file` instead?

**Comment by BertalanD** [comment]
Closing, as I'd rather create a no-GUI mode for CrashReporter instead.

**Pull request closed without merging**